Orleans.Serialization.NodaTime/CalendarSystemCopier.cs

```diff
@@ -0,0 +1,19 @@
+using System;
+using NodaTime;
+using Orleans.Serialization.Cloning;
+
+namespace Orleans.Serialization.NodaTime;
+
+[RegisterCopier]
+public class CalendarSystemCopier: IDeepCopier<CalendarSystem?>, IGeneralizedCopier
+{
+    public CalendarSystem? DeepCopy(CalendarSystem? input, CopyContext context)
+    {
+        return input;
+    }
+
+    public bool IsSupportedType(Type type)
+    {
+        return type == typeof(CalendarSystem);
```

Comment:
If `CalendarSystem` is an abstract class, this should use `typeof(CalendarSystem).IsAssignableFrom(type)` instead.

Otherwise, if it's concrete and subtypes are not copyable by this implementation (eg, they are not immutable), then you should delete this and the `IGeneralizedCodec` interface

Reply:
That's actually was my assumption as well. This is the definition of `CalendarSystem`:

```cs
    [Immutable]
    public sealed class CalendarSystem
    {
```

But, If I remove `IGeneralizedCopier` there are multiple tests that fail:

```
Orleans.Serialization.CodecNotFoundException: Could not find a copier for type NodaTime.CalendarSystem.
```